### PR TITLE
Add multizone types

### PR DESCRIPTION
**Pull request by Firstyear**
This adds the multizone types from the lifx developer docs.

### CODE REVIEW

File: lifx-core/src/lib.rs [hunk 1]
```diff
@@ -1164,6 +1164,22 @@ pub enum Message {
...
+        typ: MultiZoneEffectType,
+        reserved: u16,
+        /// The time it takes for one cycle of the effect in milliseconds
+        speed: u32,
+        /// The amount of time left in the current effect in nanoseconds
+        duration: u64,
+        reserved7: u32,
+        reserved8: u32,
+        /// The parameters that was used in the request.
+        parameters: [u8; 32],
```
Comment (eminence) [inline lifx-core/src/lib.rs]: The [lifx docs](https://lan.developer.lifx.com/docs/changing-a-device#setmultizoneeffect---packet-508) say this:

> This field is **8** `4` byte fields which change meaning based on the effect that is running.

I'm wondering if it makes sense to write this field as a `[u32; 8]` ?

File: lifx-core/src/lib.rs [hunk 2]
```diff
@@ -1180,6 +1196,15 @@ pub enum Message {
         parameters: [u8; 32],
     },
 
+    /// Message type 510
+    SetExtendedColorZones {
+        duration: u64,
```
Comment (eminence) [inline lifx-core/src/lib.rs]: The [docs](https://lan.developer.lifx.com/docs/changing-a-device#setextendedcolorzones---packet-510) say that this is a `u32` not `u64`

**Review by eminence — CHANGES_REQUESTED** [review]
Thanks!  Look good except for one small datatype issue

File: lifx-core/src/lib.rs (hunk 2 above)
Comment (Firstyear) [reply lifx-core/src/lib.rs]: Ahhh indeed! I copied from the other nearby structure and that was wrong. I'll fix up from your feedback, thank you!

File: lifx-core/src/lib.rs (hunk 1 above)
Comment (Firstyear) [reply lifx-core/src/lib.rs]: Yeah, that would make more sense  Ithink.

Comment (Firstyear) [reply lifx-core/src/lib.rs]: Would you want parameters in StateMultiZoneEffect updated to match?

**Review by eminence — APPROVED** [review]

**Comment by eminence** [comment]
Thanks!

**Pull request merged**